runtime/runtime/src/state_viewer.rs

```diff
@@ -112,6 +112,11 @@ impl TrieViewer {
         // TODO(#1015): Add ability to pass public key and originator_id
         let originator_id = contract_id;
         let public_key = PublicKey::empty(KeyType::ED25519);
+        let cache = epoch_info_provider.contract_cache();
+        let cache = match &cache {
+            Some(cache) => Some((*cache).as_ref()),
+            None => None,
+        };
```

Comment:
This should be enough :thinking: 

```suggestion
        let cache = epoch_info_provider.contract_cache().as_ref();
```

Reply:
Rust compiler tends to disagree:
```
error[E0308]: mismatched types
   --> runtime/runtime/src/state_viewer.rs:160:17
    |
160 |                 cache,
    |                 ^^^^^ expected trait object `dyn near_primitives::types::CompiledContractCache`, found struct `std::sync::Arc`
    |
    = note: expected enum `std::option::Option<&dyn near_primitives::types::CompiledContractCache>`
               found enum `std::option::Option<&std::sync::Arc<dyn near_primitives::types::CompiledContractCache>>`
```

Reply:
Oh, we have Arc there, use `as_deref()` there to force Rust to go through the dereference chain instead of just `Option<T> -> Option<&T>`

```suggestion
        let cache = epoch_info_provider.contract_cache().as_deref();
```

Reply:
Had to create a temp to hold a value, but applied similar solution.